apps/front-client/src/features/courses/hooks/useGetUserCourse.ts

```diff
@@ -0,0 +1,50 @@
+import { Chapter } from '@prisma/client';
+import { useQuery } from '@tanstack/react-query';
+import { useSession } from 'next-auth/react';
+
+const COURSE_KEY = 'course';
+
+interface CouseI {
+  id?: string;
+  title?: string;
+  description?: string;
+  progress_tracking_id?: string;
+  user_id?: string;
+  is_deleted?: boolean;
+  created_by?: string;
+  updated_by?: string;
+  created_at?: Date;
+  updated_at?: Date;
+  chapters: Chapter[];
+}
+export const fetchUserCourse = async (
+  id: string,
+  access_token?: string
+): Promise<CouseI> => {
+  const url = new URL(`/api/course/${id}`, process.env.API_URL);
+
+  const response = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${access_token}`,
+    },
+  });
+  return response.json();
+};
+
+export const useGetUserCourse = (id: string) => {
+  const { data } = useSession();
+
+  const access_token = data?.access_token;
+
+  const query = useQuery({
+    queryKey: [COURSE_KEY],
+    queryFn: () => fetchUserCourse(id, access_token),
+    refetchOnWindowFocus: false,
```

Comment:
Pour pas que le hook se lance tant qu'on à pas l'id :)
```suggestion
    refetchOnWindowFocus: false,
    enable: !!id,
```

Reply:
Bien vu

Reply:
Mais pas grave ca va rester comme ça puisque ça déclenche : No overload matches this call.
  The last overload gave the following error.
    Object literal may only specify known properties, and 'queryKey' does not exist in type 'readonly unknown[]'.ts(2769)

Reply:
??